src/Compilers/Core/Portable/FileSystem/PathUtilities.cs

```diff
@@ -773,6 +773,31 @@ public static bool IsValidFilePath([NotNullWhen(true)] string? fullPath)
         public static string NormalizeWithForwardSlash(string p)
             => DirectorySeparatorChar == '/' ? p : p.Replace(DirectorySeparatorChar, '/');
 
+        /// <summary>
+        /// Takes a normalized full path from <see cref="NormalizeWithForwardSlash(string)"/> and expands any '..' or '.'
```

Comment:
>or '.' [](http://example.com/codeflow?start=115&length=6)

Where are we handling "."?

Reply:
It looks like "." is handled in `GetPathParts()`. Consider adding a comment with the call to `GetPathParts()` below.